### PR TITLE
feat: implement `negate`

### LINKED ISSUE #72: [Feature]: `unary` for fixed-point math lib
### What is the feature you would like to see?

Implement checked unary.

### PR DESCRIPTION
**Pull request by immrsd**
- Fixes #72
- The function was initially called `unary` in the task, but was implemented as `negate`

## Summary by CodeRabbit

* **New Features**
  * Introduced unary negation operation for SD29x9 fixed-point numbers, enabling users to compute the negative value of any number in the system.

* **Tests**
  * Comprehensive test coverage added for the negation operation, verifying correct behavior across standard operations, edge cases, boundary values, and special scenarios.

### CODE REVIEW

**Review by ericnordelo — APPROVED** [review]
Excellent work @immrsd. I left a small suggestion for your consideration but it looks ready.

**Review by bidzyyys — APPROVED** [review]
LGTM!

File: math/fixed_point/sources/sd29x9/sd29x9_base.move [hunk 1]
```diff
@@ -120,6 +120,16 @@ public fun mod(x: SD29x9, y: SD29x9): SD29x9 {
     wrap_components(Components { neg: x_components.neg, mag: remainder })
 }
 
+/// Implements unary negation operation (-x) for SD29x9 type.
+public fun negate(x: SD29x9): SD29x9 {
+    let value = decompose(x.unwrap());
+    if (value.mag == 0) {
+        sd29x9::zero()
+    } else {
```
[the remark this answers is not in the document]
Comment (immrsd) [reply math/fixed_point/sources/sd29x9/sd29x9_base.move]: Makes sense! In case of 0 the call will now additionally go through the `wrap_components` func, but for all other values there will be single comparison with zero instead of two
